addon-test-support/@percy/ember/index.js

```diff
@@ -0,0 +1,128 @@
+let isPercyRunning = true;
+// Capture fetch before it's mutated by Pretender
+let PercyFetch = window.fetch;
+
+function envInfo() {
+  let frameworkVersion = () => {
+    if (window.QUnit) {
+      return `qunit/${window.QUnit.version}`;
+    } else if (window.Mocha) {
+      // Doesn't look easy to grab the version while in the browser
+      return `mocha/unknown`;
+    }
+
+    return 'unknown';
+  };
+
+  return `ember/${window.Ember.VERSION}/${frameworkVersion()}`;
+}
+
+function clientInfo() {
+  return `@percy/ember/v2.0.0`;
+}
+
+// This will only remove the transform applied by Ember's defaults
+// If there are custom styles applied, use Percy CSS to overwrite
+function removeEmberTestStyles(dom) {
+  dom
+    .querySelector('#ember-testing')
+    .setAttribute(
+      'style',
+      'width: initial !important; height: initial !important; transform: initial !important; zoom: initial !important;'
+    );
+}
+
+export default async function percySnapshot(name, options = {}) {
+  // Skip if Testem is not available (we're probably running from `ember server` and Percy is not
+  // enabled anyway).
+  if (!window.Testem) {
+    return;
+  }
+
+  if (!isPercyRunning) {
+    return false;
+  }
+
+  function setSnapshotName() {
+    // Automatic name generation for QUnit tests by passing in the `assert` object.
+    if (name.test && name.test.module && name.test.module.name && name.test.testName) {
+      name = `${name.test.module.name} | ${name.test.testName}`;
+    } else if (name.fullTitle) {
+      // Automatic name generation for Mocha tests by passing in the `this.test` object.
+      name = name.fullTitle();
+    }
+  }
+
+  async function fetchDOMLib() {
```

Comment:
Since we probably only want to do this once, we could define this outside of the function and make it cachable.